File: examples-sdk/csharp/008-basic-credit-quality-monitor/Main.cs
// 008-basic-credit-quality-monitor — C# SDK equivalent
// Compile: vil compile --from csharp --input 008-basic-credit-quality-monitor/Main.cs --release

#r "sdk/csharp/Vil.cs"

var p = new VilPipeline("CreditQualityMonitorPipeline", 3082);
p.Sink("quality_monitor_sink", 3082, "/quality-check");
p.Source("quality_credit_source", "json");
p.Route("sink.trigger_out", "source.trigger_in", "LoanWrite");
p.Route("source.response_data_out", "sink.response_data_in", "LoanWrite");
p.Route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
p.Compile();

File: examples-sdk/java/008-basic-credit-quality-monitor/Main.java
// 008-basic-credit-quality-monitor — Java SDK equivalent
// Compile: vil compile --from java --input 008-basic-credit-quality-monitor/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilPipeline p = new VilPipeline("CreditQualityMonitorPipeline", 3082);
        p.sink(3082, "/quality-check", "quality_monitor_sink");
        p.source("", "json", "quality_credit_source");
        p.route("sink.trigger_out", "source.trigger_in", "LoanWrite");
        p.route("source.response_data_out", "sink.response_data_in", "LoanWrite");
        p.route("source.response_ctrl_out", "sink.response_ctrl_in", "Copy");
        p.compile();
    }
}
